### PR TITLE
Add feature flags: buffering & strings as files

### PR DESCRIPTION
**Pull request by smheidrich**
See https://github.com/daggaz/json-stream/issues/45#issuecomment-1616727501

### CODE REVIEW

File: json_stream_rs_tokenizer/__init__.py [hunk 1]
```diff
@@ -60,15 +60,42 @@ class RequestedFeatureUnavailable(ExtensionException):
...
         requires_bigint: Deprecated, has no effect as arbitrary-size
-        integers are now always supported via fallback to conversion in Python.
+            integers are now always supported via fallback to conversion in
+            Python.
+        requires_buffering: Whether user-defined buffering is required. Not
+            supported at all yet.
+        requires_strings_as_streams: Whether the ability to return strings as
+            streams is required. Not supported at all yet.
+        kwargs: Any other required features that this version of
+            json-stream-rs-tokenizer may not even be aware of yet.
```
Comment (smheidrich) [inline json_stream_rs_tokenizer/__init__.py:79]: @daggaz Do you think this API is fine or would it be better to have a parameter like `required_features` that takes the names of parameters like `buffering` or `strings_as_files` directly? Maybe the latter makes the code simpler on the json-stream side? E.g. something like

```python
def default_tokenizer(..., buffering=-1, strings_as_files=False):
    kwargs = dict(..., buffering=buffering, strings_as_files=strings_as_files)
    try:    
        tokenizer = rust_tokenizer_or_raise(required_features=kwargs.values())
        tokenizer(**kwargs)
    except:
        ...
        return tokenize(**kwargs)
```

would be possible, whereas with the `requires_*` parameters you always have to translate from raw kwargs to those first...

Comment (daggaz) [reply json_stream_rs_tokenizer/__init__.py:79]: I agree, I've updated [the PR](https://github.com/daggaz/json-stream/pull/46) for the buffering changes in the json-stream repo with a proposed new API.

You can probably see it came with some other related changes.

I like the idea of passing the full arguments with values. It allows in the future for things to be variant on the argument value itself (for example, something like `some_mode=SOME_CONSTANT` where you may or may not support all the available constants).

If this works for you, great! If you'd like to keep backwards compatibility with the existing `requires_bigint` flag then we'll need a rethink.

Comment (smheidrich) [reply json_stream_rs_tokenizer/__init__.py]: Thanks, I think your proposed interface is better and have changed the PR accordingly. I don't think leaving `requires_bigint` in for backwards-compatibility changes anything about this, it merely looks inconsistent but doesn't conflict with the new API in any way, considering the latter is keyword-only (and assuming there will never be a tokenizer parameter called `requires_bigint`, which seems like a given).

**Review by daggaz — APPROVED** [review]
Looks good to me!

**Comment by smheidrich** [comment]
@daggaz This is available in 0.4.21 now.

**Pull request merged**